packages/commonwealth/client/scripts/views/components/reaction_button/ReactionButtonLogic.tsx

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useState } from 'react';
+import TopicGateCheck from '../../../controllers/chain/ethereum/gatedTopic';
+import type { ChainInfo, Thread } from '../../../models/index';
+import app from '../../../state';
+
+export const ReactionButtonLogic = (thread: Thread, setReactors) => {
```

Comment:
Good that you extracted the logic, like it, but this should be hook, not component.

Change the file name to `useReactionButton` and name the hook the same way.

then use it like this in other components
```
const {
    dislike,
    hasReacted,
    isLoading,
    isUserForbidden,
    like,
    setIsLoading,
  } = useReactionButton()
  ```